### PR TITLE
CORE-667 - evm sdk package

### PR DESCRIPTION
**Pull request by AlekseiPetrovDev**
https://dialect.atlassian.net/browse/CORE-667

### CODE REVIEW

**Review by tsmbl — COMMENTED** [review]
Overall lgtm, several small comments

File: packages/blockchain-sdk-evm/src/wallet-adapter/dialect-evm-wallet-adapter.interface.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import type {
+  TransactionConfig,
+  SignedTransaction,
+  EncryptedKeystoreV3Json,
+  Sign,
+} from 'web3-core';
```
Comment (zhelezkov) [inline packages/blockchain-sdk-evm/src/wallet-adapter/dialect-evm-wallet-adapter.interface.ts:6]: should we install `web3-core` in devDeps? Cause I saw only `web3`

File: packages/blockchain-sdk-evm/src/auth/evm-ed25519-authentication-facade-factory.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  get(): AuthenticationFacade {
+    return new AuthenticationFacade(
+      this.tokenSigner,
+      new DefaultTokenGenerator(this.tokenSigner),
+      EvmEd25519AuthenticationFacadeFactory.createAuthenticator(),
+    );
+  }
+}
```
Comment (zhelezkov) [inline packages/blockchain-sdk-evm/src/auth/evm-ed25519-authentication-facade-factory.ts:32]: this is def a proven to work implementation suitable for our backend. But EVM has it's own specification describing how to implement web2 authentication. I'm not insisting to change it rn, but still good to consider and be aware of pros and cons in comparison to the current implementation
https://eips.ethereum.org/EIPS/eip-4361

@tsmbl wdyt?

File: packages/blockchain-sdk-evm/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    "@dialectlabs/sdk": "^1.0.0",
+    "@ethereumjs/evm": "^1.2.2"
+  },
+  "peerDependencies": {
+    "@dialectlabs/sdk": "1.x",
+    "@ethereumjs/evm": "^1.x"
+  },
+  "dependencies": {
+    "@ethereumjs/evm": "^1.2.2",
+    "web3": "^1.8.1"
```
[the remark this answers is not in the document]
Comment (AlekseiPetrovDev) [reply packages/blockchain-sdk-evm/package.json:36]: to be honest I missed ethers.js... but just checked this lib and looks like ether.js have more benefits then web3. At least smaller size and should support for eth all cases. Let me check tomorrow morning, possibly worth switch to ethers.js

File: packages/blockchain-sdk-evm/src/auth/evm-ed25519-token-validator.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { Token, TokenHeader, TokenValidator } from '@dialectlabs/sdk';
+import Web3 from 'web3';
+import { EVM_ED25519_TOKEN_SIGNER_ALG } from './evm-ed25519-token-signer';
+
+export class EvmEd25519TokenValidator extends TokenValidator {
+  isSignatureValid(token: Token): boolean {
+    const messageBase64 = token.base64Header + '.' + token.base64Body;
+    const message = `0x${Buffer.from(messageBase64, 'utf-8').toString('hex')}`;
+
+    const web3 = new Web3()
```
Comment (zhelezkov) [inline packages/blockchain-sdk-evm/src/auth/evm-ed25519-token-validator.ts:10]: could you also please configure prettier in your IDE and format all files

**Review by zhelezkov — CHANGES_REQUESTED** [review]

File: packages/blockchain-sdk-evm/src/auth/evm-ed25519-authentication-facade-factory.ts (hunk 2 above)
Comment (tsmbl) [reply packages/blockchain-sdk-evm/src/auth/evm-ed25519-authentication-facade-factory.ts:32]: Thanks for bringing this spec up! I def think this paper makes a lot of sense so I would def consider it to be aware about pros and cons in comparison to the current implementation, also there's super-useful security section. As I understand this EIP is in still in review stage, so I think we need to study it and be aware, but probably hold spec implementation for now
I personally think we will need to revisit token structure for all chains at some point, not only for evm
1. enable human readable data in message signing (as I remember it was discussed some time ago)
2. take into account federated environments / sso with multiple backends owned by different parties
3. smth else?

wdyt? @zhelezkov @AlekseiPetrovDev

Comment (zhelezkov) [reply packages/blockchain-sdk-evm/src/auth/evm-ed25519-authentication-facade-factory.ts:32]: sg for me

File: packages/blockchain-sdk-evm/src/wallet-adapter/dialect-evm-wallet-adapter.interface.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+import type { Bytes } from 'ethers/lib/utils';
```
Comment (zhelezkov) [inline packages/blockchain-sdk-evm/src/wallet-adapter/dialect-evm-wallet-adapter.interface.ts:1]:
```suggestion
import type { Bytes } from 'ethers';
```

**Review by zhelezkov — APPROVED** [review]
lgtm

**Review by tsmbl — APPROVED** [review]
lgtm

**Review by AlekseiPetrovDev — COMMENTED** [review]
lgtm

**Comment by gklawcode** [comment]
lgtm

**Review by gklawcode — APPROVED** [review]

**Pull request merged**